lib/src/SlashRemover.cc

```diff
@@ -0,0 +1,93 @@
+#include "drogon/plugins/SlashRemover.h"
+#include "drogon/HttpAppFramework.h"
+#include "drogon/utils/FunctionTraits.h"
+#include <functional>
+#include <string>
+
+using namespace drogon;
+using namespace plugin;
+using std::string;
+
+#define TRAILING_SLASH_REGEX ".+\\/$"
+#define DUPLICATE_SLASH_REGEX ".*\\/{2,}.*"
+
+static inline bool removeTrailingSlashes(string& url)
+{
+    // If only contains '/', it will return -1, and added with 1 it will be 0,
+    // hence empty, meaning root
+    url.resize(url.find_last_not_of('/') + 1);
+    if (url.empty())  // Root
+    {
+        url = '/';
+        return true;
+    }
+    return false;
+}
+static inline void removeDuplicateSlashes(string& url)
+{
+    size_t a = 1, len = url.size();
+    for (; a < len && (url[a - 1] != '/' || url[a] != '/'); ++a)
+    {
+    }
```

Comment:
```suggestion
    for (; a < len && (url[a - 1] != '/' || url[a] != '/'); ++a);
```

Did you need an empty block? Can we do this instead?

Reply:
It was that way before, but g++ complained with a warning, so I gave it an empty body to suppress the warning

Reply:
After the new commit it no longer complains, but the formatter puts the semicolon on a new line